li-protobuf/src/main/java/com/linkedin/data/protobuf/TextBuffer.java

```diff
@@ -0,0 +1,73 @@
+/*
+   Copyright (c) 2020 LinkedIn Corp.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+
+package com.linkedin.data.protobuf;
+
+/**
+ * A container for holding text data in the form of a char[] to minimize allocations when parsing strings.
+ */
+final class TextBuffer
```

Comment:
It is better to have some unit test for this class heavily used here

Reply:
This class is implictly tested by Protobuf reader classes since it is used in reading strings. Same for Utf8Utils class.